rusk-wallet/src/lib/prompt.rs

```diff
@@ -265,69 +273,135 @@ pub(crate) fn choose_command(offline: bool) -> Option<PromptCommand> {
+            let cmd = Cli::Stake {
                 key,
-                stake_key,
-                amt,
-                gas_limit,
-                gas_price,
+                stake_key: request_key_index("stake"),
+                amt: request_token_amt("stake", balance),
+                gas_limit: request_gas_limit(),
+                gas_price: Some(request_gas_price()),
+            };
+            match confirm(&cmd) {
+                true => Some(cmd),
+                false => None,
             }
         }
         // Withdraw stake
         Prompt::Withdraw(key) => {
-            let stake_key = request_key_index("stake");
-            let gas_limit = request_gas_limit();
-            let gas_price = Some(request_gas_price());
-            Cli::WithdrawStake {
+            let cmd = Cli::WithdrawStake {
                 key,
-                stake_key,
-                gas_limit,
-                gas_price,
+                stake_key: request_key_index("stake"),
+                gas_limit: request_gas_limit(),
+                gas_price: Some(request_gas_price()),
+            };
+            match confirm(&cmd) {
+                true => Some(cmd),
+                false => None,
             }
         }
         // Export BLS Key Pair
-        Prompt::Export => {
-            let key = request_key_index("stake");
-            let encrypt = confirm_encryption();
-            Cli::Export {
-                key,
-                plaintext: !encrypt,
-            }
+        Prompt::Export => Some(Cli::Export {
+            key: request_key_index("stake"),
+            plaintext: !confirm_encryption(),
+        }),
+    }
+}
+
+/// Request user confirmation for a trasfer transaction
+fn confirm(cmd: &CliCommand) -> bool {
+    use CliCommand as Cli;
+    match cmd {
+        Cli::Transfer {
+            key: _,
+            rcvr,
+            amt,
+            gas_limit,
+            gas_price,
+        } => {
+            let max_fee = gas_limit * gas_price.unwrap();
+            println!(
+                "   > Recipient = {}..{}",
+                &rcvr[..10],
+                &rcvr[rcvr.len() - 11..]
+            );
+            println!("   > Amount to transfer = {} Dusk", to_dusk(amt));
+            println!("   > Max fee = {} Dusk", to_dusk(&max_fee));
+            ask_confirm()
         }
+        Cli::Stake {
+            key: _,
+            stake_key,
+            amt,
+            gas_limit,
+            gas_price,
+        } => {
+            let max_fee = gas_limit * gas_price.unwrap();
+            println!("   > Stake key = {}", stake_key);
+            println!("   > Amount to stake = {} Dusk", to_dusk(amt));
+            println!("   > Max fee = {} Dusk", to_dusk(&max_fee));
+            ask_confirm()
+        }
+        Cli::WithdrawStake {
+            key: _,
+            stake_key,
+            gas_limit,
+            gas_price,
+        } => {
+            let max_fee = gas_limit * gas_price.unwrap();
```

Comment:
and here

Reply:
Safe, safe, safe! There's always a default value given to the `Option`

Reply:
and here, here, here